node/silkworm/db/stages.hpp

```diff
@@ -45,6 +45,9 @@ inline constexpr const char* kIntermediateHashesKey{"IntermediateHashes"};
 // Apply Keccak256 to all the keys in the state
 inline constexpr const char* kHashStateKey{"HashState"};
 
+// Generating history index for accounts
+inline constexpr const char* kHistoryIndexKey{"HistoryIndex"};
```

Comment:
Why add "HistoryIndex"? There exist "AccountHistoryIndex" & "StorageHistoryIndex" already.

Reply:
Cause `HistoryIndex` is referred to whole stage which, in reality, is divided in two substages i.e. AccountHistoryIndex and StorageHistoryIndex. 
`HistoryIndex` is only updated when **both** substages are completed: as a result we can interrupt the stage and **IF** AccountHistory has already been committed, it restarts from StorageHistoryIndex instead of redoing everything.
Basically is relevant for initial from-scratch cycles when the stage duration is in hours.

Reply:
Imho is an improvement from Erigon where instead the whole history is committed all-or-nothing (both accounts and storage)

Reply:
Perhaps then it's better to explicitly have two stages, AccountHistoryIndex and StorageHistoryIndex?

Reply:
That would require to have two separate stages classes and move out common code in a third shared class which does the actual work. Imho not worth ... but let you decide

Reply:
Let's skip it for now and revisit later.